manager/runner.go

```diff
@@ -686,7 +691,23 @@ func (r *Runner) runTemplate(tmpl *template.Template, runCtx *templateRunCtx) (*
 		Env:   r.childEnv(),
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, tmpl.Source())
+		if tmpl.ErrFatal() {
+			return nil, errors.Wrap(err, tmpl.Source())
+		}
+		log.Printf("[ERR] (runner) %s: %v", tmpl.Source(), err)
+		event.Error = err
+
+		if lastEvent != nil {
+			// Keep watching our dependencies so that we retry when they update.
+			for _, d := range lastEvent.UsedDeps.List() {
+				if _, ok := runCtx.depsMap[d.String()]; !ok {
+					runCtx.depsMap[d.String()] = d
+				}
+			}
+			event.UsedDeps = lastEvent.UsedDeps
```

Comment:
The whole handling of event & lastEvent in error cases could probably use extra scrutiny. Everything seems to work fine, but there could be something I missed.

Reply:
This isn't the prettiest thing in the world but, given the code it must interact with, it is probably as good as it can be. In other words, I think this works fine. It should also get refactored as part of the work to port consul-template to use the hcat library, as that library manages dependencies differently.  I *think* it won't even be needed.

Anyways... extra scrutiny given and :+1:.